transaction/summary/src/lib.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+#![no_std]
+
+extern crate alloc;
+
+#[cfg(feature = "mc-account-keys")]
+mod data;
+mod error;
+mod report;
+mod verifier;
+
+#[cfg(feature = "mc-account-keys")]
+pub use data::{verify_tx_summary, TxOutSummaryUnblindingData, TxSummaryUnblindingData};
+pub use error::Error;
+pub use report::{TransactionEntity, TxSummaryUnblindingReport};
+pub use verifier::{TxSummaryStreamingVerifier, TxSummaryStreamingVerifierCtx};
```

Comment:
in ring signature we have the `internals` flag to gate access to streaming primitives, could repeat this here for `TxSummaryStreamingVerifierCtx`?

Reply:
I'm a bit skeptical of `internals` flags -- either a piece of API is public, or it isn't. Cynically, an internals flag is kind of like a weird middle ground to avoid making a decision. I think we should just say that, these types are public, and simply document that most users won't need them.

Reply:
i think the justification wrt. the ring signer was that you can misuse the streaming version in ways that the higher level API precludes, like misconstructing a ring or a hash is easier with the streaming version, but okay either way

Reply:
okay, i see that.

Another drawback of actually having a `cargo` feature called `internals` is that it increases your testing surface area. Now you probably want to test that the code builds with the flag on and off.

In other projects, like prost for instance, the way they deal with internals is they use a `#[doc(hidden)]` attribute, so it's just not documented.

In serde, they just have a module called `serde::__private` and the internals are there.

Neither of these creates an additional cargo feature, so it dodges that particular problem

Reply:
My bias would be, if it's more possible to misconstruct a ring with a given API, let's just document that fact, and skip the "internals" feature flag

Reply:
Sounds like a job for `"hazmat"`?